lib/slack/rtm.ex

```diff
@@ -10,15 +10,26 @@ defmodule Slack.Rtm do
   @moduledoc false
 
   def start(token) do
-    url = Application.get_env(:slack, :url, "https://slack.com") <> "/api/rtm.start?token="
+    slack_url(token)
+    |> HTTPoison.get()
+    |> handle_response()
+  end
 
-    case HTTPoison.get(url <> token) do
-      {:ok, %HTTPoison.Response{body: body}} ->
-        case JSX.decode(body, [{:labels, :atom}]) do
-          {:ok, json}       -> {:ok, json}
-          {:error, reason}  -> {:error, %JSX.DecodeError{reason: reason, string: body}}
-        end
-      {:error, reason} -> {:error, reason}
+  defp handle_response({:ok, %HTTPoison.Response{body: body}}) do
+    with {:ok, json} <- JSX.decode(body, [{:labels, :atom}]),
```

Comment:
Hmm, it seems like Slack will rate limit you and send back text, which is accounted for here https://github.com/BlakeWilliams/Elixir-Slack/blob/master/lib/slack/bot.ex#L51

Just want to make sure you still think its appropriate to crash in this case.